cmake/Linters.cmake

```diff
@@ -83,7 +80,49 @@ if(NOT ${CLANG_FORMAT_CMD})
     )
   endif()
 else()
-  message(WARNING "clang-format executable not found. Disabling clang-format checks. clang-format can be installed with \"pip3 install --user --upgrade clang-format\"")
+  message(WARNING "clang-format executable not found. Disabling clang-format checks. clang-format can be installed with \"pip3 install --user --upgrade clang-format\" or your distribution's package manager.")
+endif()
+
+
+#################### CLANG-TIDY
+
+# TODO: Add files here once clang-tidy is fast enough.
+# file(
+#   GLOB_RECURSE P4C_CLANG_TIDY_LINT_LIST  FOLLOW_SYMLINKS
+# )
+# add_clang_tidy_files(${P4C_SOURCE_DIR} "${P4C_CLANG_TIDY_LINT_LIST}")
```

Comment:
The problem is that, until we have figured out this slowness issue, we can only apply clang-tidy selectively. If we added all the cpplint files the amount we would have to check would be overwhelming, when you just want to check a local set of files in your target. 

I have been spot-checking files by adding `add_clang_tidy_files(${P4C_SOURCE_DIR} "${P4C_LINT_LIST}")` for a smaller subset of code I am interested in. I have not yet contributed these automated fixes because I am still tweaking rules etc.

Reply:
Hmm, this is also made worse by using the `xargs` which prevents running `clang-tidy` in parallel. This can be side-stepped by the following construction:

* Use empty `add_custom_target` to add the actual target `clang-tidy`.
* Use `add_custom_command(TARGET clang-tidy POST_BUILD COMMAND ...` for each file

But I still wonder if there is a better way of doing this then just manually adding files one wants to check (I would not expect clang-tidy to get significantly faster any time soon...). How long does it take to run clang-tidy over the all the CPPLINT-linted files?

Reply:
The xargs is necessary because of a shell quirk. At some point the number of files as arguments grew too large. 
There is a [parallel runner](https://github.com/llvm/llvm-project/blob/main/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py) but not sure it helps much when it takes ~10-30 second to analyze a single file. 

I think the underlying problem is that clang-tidy checks system-files by default. Once we find a workaround for that it should be significantly faster. 

Currently, checking all the CPPLINT files would a day at least.